ingest/tiger_slack/import.py

```diff
@@ -203,45 +203,89 @@ async def insert_messages(pool: AsyncConnectionPool, content: str) -> None:
             raise
 
 
-@logfire.instrument("load_messages", extract_args=["directory"])
-async def load_messages(pool: AsyncConnectionPool, directory: Path) -> None:
+async def process_file_worker(
+    pool: AsyncConnectionPool,
+    file_queue: asyncio.Queue[tuple[str, Path] | None],
+    worker_id: int,
+) -> None:
     buffer = ""
     item_count = 0
-    async for channel_id, file, content in channel_files(pool, directory):
-        num_elements = len(re.findall(r"^    {$", content, re.MULTILINE))
-        content = re.sub(
-            r"^    {$",
-            f'    {{\n        "channel_id": "{channel_id}",',
-            content,
-            flags=re.MULTILINE,
-        )
-        if len(buffer) == 0:
-            buffer = content
-        else:
-            buffer = buffer.rstrip("]") + "," + content.lstrip("[")
-        item_count += num_elements
-        if item_count >= 500:
-            with logfire.span(
-                "loading_messages_batch",
-                channel_id=channel_id,
-                file=file,
-                num_messages=item_count,
-            ):
-                await insert_messages(pool, buffer)
-            buffer = ""
-            item_count = 0
 
+    while True:
+        item = await file_queue.get()
+        if item is None:  # Sentinel value to stop worker
+            break
+
+        channel_id, file = item
+
+        try:
+            # Read file asynchronously using aiofiles
+            async with aiofiles.open(file, mode="r") as f:
+                content = await f.read()
```

Comment:
While it's possible that if we were to read all files contents into `file_queue` would be faster, I think the memory overhead would not work, especially for any sort of larger org, so we instead read it on the fly. Given the highly async nature of this setup, went with using `aiofiles` vs `Pathlib` so as to avoid anything blocking as much as possible.